torch/csrc/autograd/python_variable.cpp

```diff
@@ -2007,9 +2025,10 @@ void THPVariable_subclass_dealloc(PyObject* self) {
 // TAGGED_BY_US or MAYBE_UNINITIALIZED; in other cases, you know where
 // var came from and can directly assert that it's DEFINITELY_UNINITIALIZED.
 // It's ALWAYS safe (albeit slower) to call this with MAYBE_UNINITIALIZED.
+template <typename T>
 static PyObject* THPVariable_NewWithVar(
     PyTypeObject* type,
-    const at::TensorBase& _var,
+    T&& _var,
```

Comment:
Oh yeah, I meant to have both the overloads you had before and dispatch to a helper function that perfect forwarded the args, the template matching /SFINAE may cause issues.

Reply:
See THPVariable_Wrap_impl

Reply:
THPVariable_NewWithVar is a static function before, so it's safe to turn it into template.